interp/src/interpreter/interpret_group.rs

```diff
@@ -55,17 +58,41 @@ fn interp_assignments<'a, I: Iterator<Item = &'a ir::Assignment>>(
         for assignment in &assigns {
             if eval_guard(&assignment.guard, &env) {
                 //first check nothing has been assigned to this destination yet
-                if assigned_ports.contains(&get_const_from_rrc(&assignment.dst))
-                {
+                let pa =
+                    PortAssignment::new(&*assignment.dst.borrow(), assignment);
+                if assigned_ports.contains(&pa) {
+                    let mut orig_str = vec![];
+                    let mut conf_str = vec![];
+
+                    let curr_assign = &pa.get_assignment();
+
+                    ir::IRPrinter::write_assignment(
+                        assigned_ports.get(&pa).unwrap().get_assignment(),
+                        0,
+                        &mut orig_str,
+                    )?;
+                    let s_orig = String::from_utf8(orig_str)
+                        .expect("Found invalid UTF-8");
```

Comment:
You might consider factoring out this repeated code into a function, like `assignment_to_string` or something.

Reply:
+1 to this. The low level `vec!` allocation and transformation into String should be wrapped into a method.